graphql/tests/query.rs

```diff
@@ -772,3 +733,36 @@ fn first_is_nullable() {
         )]))
     );
 }
+
+#[test]
+fn nested_variable() {
+    let query = graphql_parser::parse_query(
+        "
+        query musicians($name: String) {
+          musicians(first: 100, where: { name: $name }) {
+            name
+          }
+        }
+    ",
+    )
+    .expect("invalid test query");
+
+    let result = execute_query_document_with_variables(
+        query,
+        Some(QueryVariables::new(HashMap::from_iter(
+            vec![(String::from("name"), q::Value::String("Lisa".to_string()))].into_iter(),
```

Comment:
What kind of value would we translate this to before? I wonder if we should use a `BigInt` here to really verify that this PR fixes the original issue.

Reply:
Before this, `$name` would be `Value::Variable` after coercion. So it really has nothing to do with the specific type. I tested manually for `BigInt`. I'm feeling slightly lazy about it but I can also modify the test data here to include some `BigInt` if you feel it's worth it.

Reply:
It's ok, let's move on :)